### PR TITLE
Make scoring only possible with a higher amount of players on server

### PR DESCRIPTION
**Pull request by ResamVi**
Would solve #135

### CODE REVIEW

**Comment by ResamVi** [comment]
A better solution would be to add a column `PlayerCount` to `s_infc_RoundScore` and when a winner of the of-the-day-challenge is determined `WHERE TableScore.PlayerCount > 8`

Need to tell me what i should rather do

**Comment by ResamVi** [comment]
@teoman002 maybe you decide which solution because you tagged me or @bretonium knows if changing the database schema causes trouble

**Comment by teoman002** [comment]
solution 1 seems simple, so I suggest to do that. We can improve the code later™.

File: src/engine/server/server.cpp [hunk 1]
```diff
@@ -4218,7 +4218,20 @@ class CSqlJob_Server_SendPlayerStatistics : public CSqlJob
...
 	{
 		char aBuf[512];
-				
+		
+		int PlayerCount = 0;
+		for(int i=0; i<MAX_CLIENTS; i++)
+		{
+			if(m_apPlayers[i] && m_apPlayers[i]->m_IsInGame == true)
+			{
+				if (std::find(vec.begin(), vec.end(), i) == vec.end())
```
Comment (bretonium) [inline src/engine/server/server.cpp]: sorry, but what is `vec` here? What does this line do?

Comment (bretonium) [reply src/engine/server/server.cpp]: > vec contains all spectators.

And where is `vec` defined and initialized?

Comment (bretonium) [reply src/engine/server/server.cpp]: even worse, if `vec` is not defined, how does it even compile?

**Comment by bretonium** [comment]
> changing the database schema causes trouble

By itself it doesn't, but we do not have the necessary infrastructure to automate schema migration, and asking administrators to do the migration manually, via .sql file, is a pain. We could add a check that the migration was applied and exit if it wasn't, but it is too much hassle for a simple issue.

File: src/engine/server/server.cpp (hunk 1 above)
Comment (teoman002) [reply src/engine/server/server.cpp]: CServer::GetActivePlayerCount() provides access to active players, there is no need to access this vec. And yes it was not defined before. It is defined in GetActivePlayerCount()

If (GetActivePlayerCount() > ... ) then do smth

**Comment by ResamVi** [comment]
> 
> 
> right, it does not compile

Yes, problem was I used some fields that only #138 had declared yet. I'll try fix it and test it as well

File: src/engine/server/server.cpp (hunk 1 above)
Comment (ResamVi) [reply src/engine/server/server.cpp]: yeah forgot to use `server_sql_release` as conf. got it working with a sql database and tested it on oi2 labs

File: src/engine/server/server.cpp [hunk 2]
```diff
@@ -4218,7 +4218,10 @@ class CSqlJob_Server_SendPlayerStatistics : public CSqlJob
 	void UpdateScore(CSqlServer* pSqlServer, int ScoreType, int Score, const char* pScoreName)
 	{
 		char aBuf[512];
-				
+
+		if (m_pServer->GetActivePlayerCount() > 8)
```
Comment (bretonium) [inline src/engine/server/server.cpp]: `> 8`
LOL

File: src/engine/server/server.cpp [hunk 3]
```diff
@@ -4218,7 +4218,10 @@ class CSqlJob_Server_SendPlayerStatistics : public CSqlJob
...
+			return
```
Comment (bretonium) [inline src/engine/server/server.cpp]: I wonder what this line without `;` in the end does

**Pull request merged**